src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -230,6 +230,28 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_StorePrimaryWiFiMACAddre
     return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_GetPrimaryMACAddress(uint8_t (&buf)[8])
+{
+    memset(buf, 0, 8);
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ThreadStackMgr().GetPrimary802154MACAddress(buf) == CHIP_NO_ERROR)
+    {
+        ChipLogDetail(DeviceLayer, "Using Thread extended MAC for hostname.");
+        return CHIP_NO_ERROR;
+    }
+#endif
+    if (DeviceLayer::ConfigurationMgr().GetPrimaryWiFiMACAddress(buf) == CHIP_NO_ERROR)
+    {
+        ChipLogDetail(DeviceLayer, "Using wifi MAC for hostname");
+        return CHIP_NO_ERROR;
+    }
+    ChipLogError(DeviceLayer, "MAC is not known, using a default.");
+    uint8_t temp[6] = { 0xEE, 0xAA, 0xBA, 0xDA, 0xBA, 0xD0 };
+    memcpy(buf, temp, 6);
```

Comment:
So we are assuming the last bytes are 0 if it's a 48 bit MAC, but the size is not given to the caller. This prevents generating the host name for operational discovery to-spec:

> The target host name SHALL be constructed using one of the
> available link-layer addresses,
> such as a 48-bit device MAC address (for Ethernet and Wi‑Fi)
> or a 64-bit MAC Extended Address (for Thread)
> expressed as a fixed-length twelve-character
> (or sixteen-character) hexadecimal string,
> encoded as ASCII (UTF-8) text
>